### PR TITLE
Fix: 1221. Non-normative hints for duplicate digest-algorithms.

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

Provides non-normative feedback on duplicate digest-algorithms

### CODE REVIEW

**Comment by LPardue** [comment]
Thanks for putting this together. However, I'm not sure if all the text really helps address #1221 now that we obsoleted parameters.

An endpoint that receives a digest field with more than one `digest-algorithm` of the same type has a couple of options. Ignore it altogether (no validation); or do validation and pick the first, pick the last, pick them all, etc. Obviously if you have the same `digest-algorithm` and different `digest-values` there is a chance that your pick could be wrong and validation might fail. We already describe a similar case wrt SRI, saying `This document specifies no requirements or guidance for user agents that experience such cases.`. I think what to do is best left as an implementation choice.

I don't see the value in talking about hypotheticals. We removed parameters because of lack evidence of usage, if someone wants to go that route they can design their own solution, which is what we recommend in Section 8 already.

So in summary, I think at most all we can do is document that it is an implementation decision to pick which entries in the possible set of digests should be verified.

**Comment by ioggstream** [comment]
@LPardue I wrote those considerations thinking about multihash, which would use a single digest-algorithm token for different crypto algorithms, but I agree I went too far :) OTOH I'd save the first lines, as I think it's worth notifying them to the implementors.

Could you please `suggest` and edit to the PR, 
eg. removing https://github.com/httpwg/http-extensions/pull/1333/files#diff-de34152d297a9eb29b85b24962949ba2e309478a5c32e1d94fd4c3ee27b4584dR1117-R1142
?

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -1098,6 +1098,49 @@ or by sending multiple representation-data-digest values from which the receiver
...
+shake-128/5: 5/<\x1c?W\xda
+shake-128/10: 10/<\x1c?W\xda\xafA8_\xd6
+~~~
+
+and then encoding it using the `sf-list` notation defined in {{?I-D.ietf-httpbis-header-structure}}.
+
+~~~ example
+Digest: shake-128=(:NS88HD9X2g==: :MTAvPBw/V9qvQThf1g==:)
+~~~
+
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:1143]:
```suggestion
```

File: draft-ietf-httpbis-digest-headers.md [hunk 2]
```diff
@@ -1098,6 +1098,49 @@ or by sending multiple representation-data-digest values from which the receiver
...
+As the receiver is free to ignore any representation-data-digest value (see {{digest}}),
+this is especially true when `Digest` contains duplicate digest-algorithms.
+The semantic of the digest-algorithm could impact on the decision, though.
+
+For example, the receiver of the following field
+
+~~~ example
+Digest: sha-256=X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=,
+        sha-256=47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=
+~~~
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:1112]:
```suggestion
An endpoint might receive multiple representation-data-digest values (see {{digest}}) that use the same digest-algorithm with different or identical digest-values. For example:

~~~ example
Digest: sha-256=X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=,
        sha-256=47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=
~~~
```

A receiver is permitted to ignore any representation-data-digest value, so validation of duplicates is left as an implementation decision. Endpoints might select all, some or none of the values for checksum comparison and, based on the intersection of those results, conditionally pass or fail digest validation.

**Review by LPardue — COMMENTED** [review]
WDYT?

File: draft-ietf-httpbis-digest-headers.md [hunk 3]
```diff
@@ -1098,6 +1098,18 @@ or by sending multiple representation-data-digest values from which the receiver
...
+An endpoint might receive multiple representation-data-digest values (see {{digest}}) that use the same digest-algorithm with different or identical digest-values. For example:
+
+~~~ example
+Digest: sha-256=X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=,
+        sha-256=47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=
+~~~
+
+could consider it suspicious and ignore it, use only one of its values, or both of them.
+
+
```
Comment (ioggstream) [inline draft-ietf-httpbis-digest-headers.md:1112]:
```suggestion
A receiver is permitted to ignore any representation-data-digest value,
so validation of duplicates is left as an implementation decision.
Endpoints might select all, some or none of the values for checksum comparison and,
based on the intersection of those results, conditionally pass or fail digest validation.
```

Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:1112]: @LPardue is this it?

Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:1112]: yep

File: draft-ietf-httpbis-digest-headers.md [hunk 4]
```diff
@@ -1098,6 +1098,19 @@ or by sending multiple representation-data-digest values from which the receiver
...
+~~~ example
+Digest: sha-256=X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=,
+        sha-256=47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=
+~~~
+
+A receiver is permitted to ignore any representation-data-digest value,
+so validation of duplicates is left as an implementation decision.
+Endpoints might select all, some or none of the values for checksum comparison and,
+based on the intersection of those results, conditionally pass or fail digest validation.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:1113]:
```suggestion
based on the intersection of those results, conditionally pass or fail digest validation.

```

**Review by LPardue — APPROVED** [review]
One whitespace nit but otherwise LGTM
